internal/go_repository_cache.bzl

```diff
@@ -64,6 +64,11 @@ GOCACHE='{gocache}'
         gopath = go_path,
         gocache = go_cache,
     )
+    for name, value in ctx.attr.go_env.items():
```

Comment:
It would be more straightforward if the given dict and the cache dict were combined before formatting the file:

Something like:

```
cache_env = {
    "GOROOT": go_root,
    "GOPATH": go_path,
    "GOCACHE": go_cache,
}
for key, value in go_env.items():
    if key in cache_env:
        fail("{} cannot be set in go_env".format(key))
    cache_env[key] = value
env_content = "\n".join("{k}='{v}'\n".format(k, v) for k, v in go_env.items()]
```

Reply:
done.